### PR TITLE
Added ignored tests to catch issue with PersistenceExceptionHandlers

### PR DESCRIPTION
**Pull request by clicktravel-basharat**
A problem has been spotted in the PersistenceExceptionHandlers, I have made a failing test for it:

If a persistence exception is thrown, an attempt will be made to coerce it into a persistence exception which is handled, even if it is not the same. e.g.

1. ItemConstraintViolationException handler defined
2. OptimisticLockException thrown
3. Attempts to coerce OptimisticLockException into a ItemConstraintViolationException
4. FAIL

The issue is that the PersistenceExceptionHandler defines the `abstract handle(PersistenceException)` method and uses that when it does not find an exact match on PersistenceException type.

### CODE REVIEW

**Comment by clicktravel-steffan** [comment]
@clicktravel-basharat One query raised. I'm not really understanding this code at all.

File: cheddar/cheddar-integration-mocks/src/main/java/com/clicktravel/infrastructure/persistence/inmemory/database/InMemoryDatabaseTemplate.java [hunk 1]
```diff
@@ -163,6 +175,28 @@ private void deleteUniqueConstraints(final Item item) {
...
 
+    /**
+     * Checks the presence of a matching unique constraint for the given item property
+     * 
+     * @param item The Item for which to check unique constraints
+     * @param propertyName The name of the property for which the unique constraint should be checked
+     * @param propertyValue The value of the property which needs to be checked for presence of unique constraint
+     * @return
+     */
+    public boolean hasMatchingUniqueConstraint(final Item item, final String propertyName, final String propertyValue) {
```
Comment (clicktravel-steffan) [inline cheddar/cheddar-integration-mocks/src/main/java/com/clicktravel/infrastructure/persistence/inmemory/database/InMemoryDatabaseTemplate.java]: Is this method only used in unit tests? If so, is it possible to move it to the unit test source code tree?

Comment (clicktravel-basharat) [inline cheddar/cheddar-integration-mocks/src/main/java/com/clicktravel/infrastructure/persistence/inmemory/database/InMemoryDatabaseTemplate.java]: It depends on some internals so the best I can do is make it package private?

Comment (clicktravel-steffan) [inline cheddar/cheddar-integration-mocks/src/main/java/com/clicktravel/infrastructure/persistence/inmemory/database/InMemoryDatabaseTemplate.java]: Package scope sounds good.

**Pull request merged**